File: benchmark/2026-09-17-7677581(latest)/standard/rubrics/probes/DBG-1/typescript/dbg1.ts
// DBG-1
class Item {
    count: number;
    name: string;
    constructor(count: number, name: string) {
        this.count = count;
        this.name = name;
    }
}

const ITERATIONS: number = 40000000;
const MODULUS: number = 1000000007;

function accumulate(items: Item[], iterations: number): number {
    let total: number = 0;
    for (let i = 0; i < iterations; i++) {
        for (const item of items) {
            total = (total * 31 + item.count + item.name.length) % MODULUS;
        }
    }
    return total;
}

const items: Item[] = [];
items.push(new Item(7, "alpha"));
items.push(new Item(11, "bravo"));
items.push(new Item(13, "charlie"));
const checksum: number = accumulate(items, ITERATIONS);
console.log("checksum=" + checksum);

File: benchmark/2026-09-17-7677581(latest)/standard/rubrics/probes/DBG-1/rust/dbg1.rs
// DBG-1
const ITERATIONS: i64 = 300000000;
const MODULUS: i64 = 1000000007;

struct Item {
    count: i64,
    name: String,
}

fn accumulate(items: &Vec<Item>, iterations: i64) -> i64 {
    let mut total: i64 = 0;
    for _i in 0..iterations {
        for item in items.iter() {
            total = (total * 31 + item.count + item.name.len() as i64) % MODULUS;
        }
    }
    total
}

fn main() {
    let mut items: Vec<Item> = Vec::new();
    items.push(Item { count: 7, name: String::from("alpha") });
    items.push(Item { count: 11, name: String::from("bravo") });
    items.push(Item { count: 13, name: String::from("charlie") });
    let checksum = accumulate(&items, ITERATIONS);
    println!("checksum={}", checksum);
}
